src/components/AlchemyRow.vue

```diff
@@ -0,0 +1,185 @@
+<template>
+  <td class="columnSpan">
+    <GameAsset class="centered"
+      :image="Assets.FromDir(`${props.group}${props.idx+1}`, 'alchemy')"
+      :height="64"/>
+  </td>
+  <td class="columnSpan">
+    <input class="columnSpan"
+      type="number" :min="0"
+      :value="alchemy.upgrades[group][props.idx]"
+      @change='customChange'/>
+  </td>
+  <td class="columnSpan">
+    <input class="columnSpan"
+      type="number" :min="alchemy.upgrades[group][props.idx]"
+      :value="alchemy.goals[group][props.idx]"
+      @change="handleAlchemyGoal($event, group, props.idx)"/>
+  </td>
+  <td class="centered columnSpan-2 effect"> 
+    {{effectChange()}}
+  </td>
+  <table class="filled">
+    <tr :style="{display: 'flex'}">
+      <td
+        :style="{width: '25%', whiteSpace: 'pre', marginRight: '5%'}" 
+        v-for="(thing, idx) in computeMaterials()"
+        :key="idx"
+      >
+        <th>{{thing[0]}}</th>
+        <tr :style="{display: 'inline-block', textAlign: 'right', width: '100%'}">
+          {{thing[1]}}
+          </tr>
+      </td>
+    </tr>
+  </table> 
+</template>
+
+
+<script lang="ts">
+import { computed, defineComponent,} from "vue";
+import { useState} from "~/State";
+import GameAsset from "~/components/GameAsset.vue";
+import { Assets, Growth } from "~/composables/Utilities";
+import { Color , Alch, Bubble} from "~/composables/Alchemy";
+import bubblesData from "~/data/bubbles.json"
+
+
+
+export default defineComponent({
+  name: "AlchemyRow",
+  components: {
+    GameAsset,
+  },
+  props: {
+    group: {
+      required: true,
+      type: String,
+    },
+    discount: {
+      required: true,
+      type: Array,
+    },
+    idx: {
+      required: true, 
+      type: Number,
+    }
+  },
+  methods: {
```

Comment:
To prevent the warning spam in the console, declare the `custom-change` event here.
```suggestion
  emits: ["custom-change"],
  methods: {
```

Warning from console for reference
```
[Vue warn]: Extraneous non-emits event listeners (customChange) were passed to component but could not be automatically inherited because component renders fragment or text root nodes. If the listener is intended to be a component custom event listener only, declare it using the "emits" option. 
  at <AlchemyRow onCustomChange=fn group="Orange" idx=3  ... > 
  at <Cauldrons> 
  at <Alchemy> 
  at <Home> 
  at <App>
warn @ runtime-core.esm-bundler.js:38
```

Reply:
This has been applied